web/src/components/NumberField.js

```diff
@@ -0,0 +1,24 @@
+import { TextField } from '@cmsgov/design-system-core';
+import React, { useEffect, useRef } from 'react';
+
+const selectTextIfZero = ({ target }) => {
+  const value = +target.value.replace(/[^0-9]/g);
+  if (value === 0) {
+    target.select();
+  }
+};
+
+const NumberField = ({ ...props }) => {
+  const textField = useRef(null);
+
+  useEffect(() => {
```

Comment:
Even though I get Hooks better now, I can't really get my head around `useEffect` or when it fires. (Which will probably change soon, because I think it's what needs to happen with the `DateField` error.) So... very nice magic you have here!

Reply:
`useEffect` is run ***after*** every render of the component. But hooks can be memoized to prevent them running _every_ time. The signature for `useEffect` is:

```js
useEffect(functionToRun, [list, of, variables, to, memoize, on]);
```

The hook will only run if one of those variables has changed. If you leave out the list of variables, it will run every time. If you pass it an empty array, it will only run the first time the component is rendered and then never again. This way, you can mimic the behavior of `componentDidMount`. The major difference between `useEffect` and `componentDidMount` is when they run - `componentDidMount` runs before render, so the `ref` here would be unset; `useEffect` runs *after* render, so `ref` is ready to go.